baybe/recommenders/meta/sequential.py

```diff
@@ -59,22 +53,16 @@ class TwoPhaseMetaRecommender(MetaRecommender):
 
     def select_recommender(  # noqa: D102
         self,
-        searchspace: SearchSpace,
+        searchspace: Optional[SearchSpace] = None,
```

Comment:
is searchspace ever needed for selecting the recommender?

Reply:
Well, yes and no:
* "No" (i.e. they shouldn't need `searchspace`) because none of our *existing* meta recommender does so (hence the `existing` in the commit message). Hence, we could technically drop it.
* But "Yes" (i.e. we should still keep the argument) because of two reasons:
  1. Selecting a recommender is technically a sub-routine of the `recommend` step, and the latter has access to `searchspace`. We discussed at some point whether or not to make the selection public, and the rationale for doing so was that selection is pretty much the core capability of a "meta"-recommender. Thus, from an interface/protocol perspective, it makes sense to keep it.
  2. In fact, I can very well imagine that future "meta"-recommenders will use it. I rather consider the `sequential` recommenders a special case in that they ignore it for their selection because they only care about order. But I can imagine dozens of other meta selectors that want to use this information. For instance, we could have "computational auto-tune recommenders" that select a recommender based on the optimization complexity, e.g. GPs for small-scale problems and Random Forests for large-scale problems. You get the point ... If we kick the argument now, we'll have a breaking change later. But if you prefer to do so, I can also live with that.

Reply:
something we encounter often: Adrians cat 🐱
- one the one hand it's `reasonable`
- on the other it's `already incorporating info from hypothetical and not immediate scenarios from the future`

Reply:
As far as I remember, we've agreed that it can stay as is in the last meeting, so will close